weekly_data_report.R

```diff
@@ -0,0 +1,131 @@
+p <- add_argument(p,'file_name',help='name of geocoded csv file')
+args <- parse_args(p)
+
+# args <- list()
+# args$file_name <- "simulated_jfs_data_geocoded.csv"
+
+# INTAKE_ID
+# SCREENING_DECISION = SCREENED OUT or SCREENED IN
+# ALLEGATION_ADDRESS, formatted in a string without punctuation, it includes city, state, and zip code
+
+message("\nNeighborhood results with < 5 instances have been censored for privacy purposes\n")
+
+suppressPackageStartupMessages(library(readr))
+suppressPackageStartupMessages(library(tidyverse))
+suppressPackageStartupMessages(library(sf))
+suppressPackageStartupMessages(library(knitr))
+
+d <- read_csv(args$file_name,
+              col_types = cols(INTAKE_ID = col_character(),
+                               SCREENING_DECISION = col_character(),
+                               DECISION_DATE = col_character(),
+                               PERSON_ID = col_character(),
+                               RACE = col_character(),
+                               ADDRESS_START = col_date(),
+                               MANDATED_REPORTER = col_character(),
+                               #  REPORTER_PERSON_ID = col_character(),
+                               address_type = col_character(),
+                               address = col_character(),
+                               bad_address = col_logical(),
+                               PO = col_logical(),
+                               lat = col_double(),
+                               lon = col_double(),
+                               score = col_double(),
+                               precision = col_character(),
+                               precise_geocode = col_logical(),
+                               fips_tract_id = col_character(),
+                               fraction_assisted_income = col_double(),
+                               fraction_high_school_edu = col_double(),
+                               median_income = col_double(),
+                               fraction_no_health_ins = col_double(),
+                               fraction_poverty = col_double(),
+                               fraction_vacant_housing = col_double(),
+                               dep_index = col_double()
+              ))
+
+###
+# d_test <- read.csv('test/simulated_jfs_data_geocoded.csv')
+# d <- d_test
+###
+d <- dplyr::mutate(d, DECISION_DATE = dht::check_dates(DECISION_DATE))
+
+# rmarkdown::render(input = '/app/aggregate_data_report.rmd',
+#                   params = list(d = d),
+#                   envir = new.env())
+
+
+tract_to_neighborhood <- readRDS('/app/tract_to_neighborhood.rds')
+neighborhood_shp <- readRDS('/app/ham_neighborhoods_dep_index_shp.rds')
+
+# Overall Summary
+date_min <- min(d$DECISION_DATE)
+date_max <- max(d$DECISION_DATE)
+
+# consider 'SCREENED IN AR' same as 'SCREENED IN'
+for (i in 1:nrow(d)) {
+  if (d$SCREENING_DECISION[i] == 'SCREENED IN AR') {
+    d$SCREENING_DECISION[i] <- 'SCREENED IN'
```

Comment:
Can we use something more straightforward here?  Like below where we census counts of 5 or less, we could do an `ifelse`.  

Or we could make a new variable that is logical to use for our purposes of dichotomizing the screening decision:

```
d %>%
  mutate(screened_in = SCREENING_DECISION %in% c("SCREENED IN", "SCREENED IN AR"))
```